### DISCUSSION TITLE
[Bug]: Issue with controller and testing

### DISCUSSION THREAD

**Discussion by pollard93** [react-hook-form/react-hook-form#6191] [Q&A]
### Version Number

7.12.2

### Codesandbox/Expo snack

https://github.com/pollard93/react-hook-form-issue

### Steps to reproduce

1. Go to and clone https://github.com/pollard93/react-hook-form-issue
2. Run `yarn && yarn test`

### Expected behaviour

There should not be an error. The error is coming from the use of the `Controller` -> `render`-> `field.onChange`. This works fine outside of a test environment. I've also tested using `setValue` with no issues.

### Relevant log output

```shell
TypeError: A is not a function
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by piers-smartwyre** [comment]
Any ideas why this is happening? Seems as though the compiled code isn't working in the jest environment as expected.

**Reply by scrungrth** [reply]
it's definitely related to this switch https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/createFormControl.ts#L924

if I use uncompiled version and hard code this `isWindowUndefined` to true, then the tests work fine.

this seems to fix it for me, definitely seems like a bug
```
jest.mock('react-hook-form', () => {
  global.window = {};
  return jest.requireActual('react-hook-form');
});
```
